### DISCUSSION TITLE
steamcmd error [solved: Docker Version >=20.10]

### DISCUSSION THREAD

**Discussion by TheoMarque** [mbround18/valheim-docker#481] [Help!] [answered]
From logs:

> valheim_1  | [ODIN][INFO]  - Installing 896660 to /home/steam/valheim
> valheim_1  | [ODIN][INFO]  - steamcmd found in path
> valheim_1  | [ODIN][INFO]  - Exited with http code: 1
> valheim_1  | cp: cannot stat '/home/steam/steamcmd/linux64/steamclient.so': No such file or directory
> valheim_1  | [Valheim][steam]: Initializing Variables....
> valheim_1  | [ODIN][ERROR] - Failed to find server executable! Please run `odin install`

Fresh image 1.4.12 default docker-compose.yml

**Comment by mbround18** [comment, marked as answer]
@TheoMarque I have seen this a couple of times in the past,

- Does the computer running the system have internet access?
- Have you tried to nuke the server folder and relaunch?
- Is the user & group Id of the user you are launching the docker compose as equal 1000?

**Reply by TheoMarque** [reply]
- Yes, is on dedicated server, with internet - new docker images are pulling (pruned all your local images).
- Sure, problem exists on fresh docker-compose place
- Yes, tested before send message [STEAM_UID 1000 - STEAM_GUID 1000] uid=1000(user) gid=1000(user) groups=1000(user),999(docker)

Tried to recognize where is a problem and I think is in steamcmd image.

**Reply by TheoMarque** [reply]
I do some additional tests, problem occur only one machine - maybe my host is banned to download some packages? How can I test it?

> valheim_1  | [Valheim][steam]: Running Install...
> valheim_1  | [ODIN][DEBUG] - Debugging set to true
> valheim_1  | [ODIN][DEBUG] - Debug mode enabled!
> valheim_1  | [ODIN][DEBUG] - Launching install command...
> valheim_1  | [ODIN][DEBUG] - Env var default 'ODIN_WORKING_DIR': '/home/steam/valheim'
> valheim_1  | [ODIN][INFO]  - Installing 896660 to /home/steam/valheim
> valheim_1  | [ODIN][DEBUG] - Env var default 'ODIN_WORKING_DIR': '/home/steam/valheim'
> valheim_1  | [ODIN][INFO]  - steamcmd found in path
> valheim_1  | [ODIN][DEBUG] - Launching install command: "/usr/bin/steamcmd" "+login anonymous" "+force_install_dir /home/steam/valheim" "+app_update 896660" "+quit"
> valheim_1  | [ODIN][INFO]  - Exited with http code: 1
> valheim_1  | cp: cannot stat '/home/steam/steamcmd/linux64/steamclient.so': No such file or directory
> valheim_1  | [Valheim][steam]: Initializing Variables....
> valheim_1  | [ODIN][DEBUG] - Debugging set to true
> valheim_1  | [ODIN][DEBUG] - Debug mode enabled!
> valheim_1  | [ODIN][DEBUG] - Launching configure command...
> valheim_1  | [ODIN][DEBUG] - Pulling config file...
> valheim_1  | [ODIN][DEBUG] - Env var found 'ODIN_CONFIG_FILE': '/home/steam/valheim/config.json'
> valheim_1  | [ODIN][DEBUG] - Config file set to: /home/steam/valheim/config.json
> valheim_1  | [ODIN][DEBUG] - Writing config file...
> valheim_1  | [ODIN][DEBUG] - Env var default 'ODIN_WORKING_DIR': '/home/steam/valheim'
> valheim_1  | [ODIN][DEBUG] - Checking env for server_executable
> valheim_1  | [ODIN][ERROR] - Failed to find server executable! Please run `odin install`

**Reply by mbround18** [reply]
Huh, while the server is running the container try accessing it via:

```sh
docker-compose exec valheim bash
```

and then in the container: 

```sh
gosu steam bash
```

Once as the steam user, try running:

```sh
odin install
```

See if it gives you some grief running manually

**Reply by TheoMarque** [reply]
I can mark as solved. A problem was in libcurl. Why? Because Debian 9.13 (marked as oldstable and with support until 06.2022) was update blocker for docker engine >19.03. So i Manually upgrade my distro to Debian Buster (10) and docker-ce to 20.10 and.. now it working as intended. Anyway, put info about docker engine version - supporting only Docker >=20

Your image was a first that exposed me a problem with docker hosting machine.
